.github/workflows/rust.yml

```diff
@@ -55,3 +55,48 @@ jobs:
     - uses: actions/checkout@v2
     - name: Run cargo fmt
       run: cargo fmt --all -- --check
+
+  publish:
+    runs-on: ubuntu-20.04
+    needs: [tests, clippy, cargo-fmt]
+    if: github.ref == 'refs/heads/master'
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Setup
+      run: |
+        git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
+        cargo install --git https://github.com/miraclx/cargo-workspaces --rev b2d49b9e575e29fd2395352e4d0df47def025039 cargo-workspaces
+
+    - name: Publish to crates.io and tag the commit responsible
+      id: version-tag-and-publish
+      run: |
+        cargo ws publish --all --yes --exact \
+          --skip-published --no-git-commit --allow-dirty \
+          --tag-existing --tag-prefix 'v' \
+          --tag-msg 'crates.io snapshot' --tag-msg $'%{\n%n: https://crates.io/crates/%n/%v}' \
+          --no-individual-tags --no-git-push
+        echo "GIT_LATEST_TAG=$(git describe --abbrev=0)" >> $GITHUB_ENV
+        echo "Latest git tag is ${{ env.GIT_LATEST_TAG }}"
+        echo "::set-output name=git_tag::$(git tag -l --format='%(body)' ${{ env.GIT_LATEST_TAG }})"
+
+    - name: Push tags to GitHub (if any)
+      run: git push --tags
+
+    - name: Extract release notes
+      id: extract-release-notes
+      uses: ffurrer2/extract-release-notes@v1
```

Comment:
Not a super huge fan of using unpinned 3rd-party actions. My understanding is that GithubActions fundamentally lack an equivalent of lockfiles for actions, so, when someone publishes a new version of an action, you are automatically upgraded to it on the next run, which is somewhat scary, given that actions have access to all kinds of secrets. For this reason, I prefer to pin actions via commit has as well ([example](https://github.com/rust-analyzer/rust-analyzer/blob/2cb85c14b622002767c66881c6a316a94e0f0be4/.github/workflows/ci.yaml#L90)), which has dubious security value, as that's a broken sha-1 hash.

No strong opinion here though.

Reply:
That's smart. I second that opinion.